frontend/src/app/analytics/Analytics.test.tsx

```diff
@@ -1,13 +1,195 @@
-import { render, screen, fireEvent } from "@testing-library/react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { act } from "react-dom/test-utils";
+import { MockedProvider } from "@apollo/client/testing";
+import createMockStore from "redux-mock-store";
+import { Provider } from "react-redux";
+import moment from "moment";
 
-import { Analytics } from "./Analytics";
+import { GetTopLevelDashboardMetricsDocument } from "../../generated/graphql";
+
+import { Analytics, EXTERNAL_DATE_FORMAT } from "./Analytics";
+
+const mockStore = createMockStore([]);
+
+const store = mockStore({
+  organization: {
+    name: "Central Schools",
+  },
+  facilities: [
+    { id: "1", name: "Lincoln Middle School" },
+    { id: "2", name: "Rosa Parks High School" },
+  ],
+});
+
+const now = moment();
+const today = now.format(EXTERNAL_DATE_FORMAT);
+
+const topLevelDashboardMetricsMockQueries = [
+  {
+    request: {
+      query: GetTopLevelDashboardMetricsDocument,
+      variables: {
+        facilityId: "",
+        startDate: moment(today, EXTERNAL_DATE_FORMAT)
+          .subtract(7, "day")
+          .hour(now.hours())
+          .minute(now.minutes())
+          .toDate(),
+        endDate: moment(today, EXTERNAL_DATE_FORMAT)
+          .hour(now.hours())
+          .minute(now.minutes())
+          .toDate(),
```

Comment:
Interestingly, I think this causes tests that were just submitted not to show up.  Submitting a test, then going to the analytics page wasn't showing the just-submitted test until (what seemed like) the next minute.

Reply:
Wow! Great catch. I'll write a helper function to include the seconds as well as hours and minutes to catch this.